### PR TITLE
ROX-18642: Move and rewrite AsyncComponent in TypeScript

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

Verified that webpack needs to see `import('componentPath')` and so cannot replace component with component path in route configuration.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

It seems more likely that change to chunks is from move than from TypeScript.

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: 418 = 3757748 - 3757330
        total: 1043 = 9901516 - 9900473
    * `ls -al build/static/js/*.js | wc -l`
        1 = 79 - 78 files
3. `yarn start` in ui

### Manual testing

Visit pages via left navigation and explore 1 (or 2 for workflow) levels of descendant pages.

### CODE REVIEW

File: ui/apps/platform/src/Containers/MainPage/AsyncComponent.tsx [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 
+type Props = Record<string, never>;
+type State = {
+    component: ElementType | null;
+};
+
 export default function asyncComponent(importComponent) {
-    class AsyncComponent extends Component {
+    class AsyncComponent extends Component<Props, State> {
+        isComponentMounted: boolean;
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/MainPage/AsyncComponent.tsx:11]: Why is this line needed?

**Review by vjwilson — APPROVED** [review]

File: ui/apps/platform/src/Containers/MainPage/AsyncComponent.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/MainPage/AsyncComponent.tsx:11]: Good question. Not obvious because the reason is in the unchanged lines below.

Declaration for `this.isComponentMounted` which is in addition to component state declared above.
